test/minecraft_tests.cpp

```diff
@@ -169,4 +169,61 @@ TEST_CASE("Test blocks struct") {
     CHECK((mc.getBlock(testLoc) == Blocks::AIR));
     mc.setBlock(testLoc, Blocks::STONE);
     CHECK((mc.getBlock(testLoc) == Blocks::STONE));
+}
+
+
+TEST_CASE("Test array unflattening utils") {
+    Coordinate loc1(0,0,0);
+    Coordinate loc2(4, 4, 4);
+
+    //block util tests
+    mc.setBlocks(loc1, loc2, Blocks::DIRT);
+    SUBCASE("getBlocks") {
+        auto expected = std::vector<std::vector<std::vector<BlockType>>>(
+            5,
+            std::vector<std::vector<BlockType>>(
+                5,
+                std::vector<BlockType>(
+                    5,
+                    Blocks::DIRT
+                )
+            )
+        );
+        std::vector<BlockType> blockArr = mc.getBlocks(loc1, loc2);
+        auto unFlattenedBlockArr = unFlattenBlocksArray(loc1, loc2, blockArr);
+        CHECK((unFlattenedBlockArr == expected));
+    }
+
+    SUBCASE("getBlocksWithData") {
+        auto expected = std::vector<std::vector<std::vector<BlockType>>>(
+            5,
+            std::vector<std::vector<BlockType>>(
+                5,
+                std::vector<BlockType>(
+                    5,
+                    Blocks::DIRT
+                )
+            )
+        );
+        std::vector<BlockType> blockArrData = mc.getBlocksWithData(loc1, loc2);
+        auto unFlattenedBlockDataArr = unFlattenBlocksArray(loc1, loc2, blockArrData);
+        CHECK((unFlattenedBlockDataArr == expected));
+    }
+
+    SUBCASE("getHeights") {
+        Coordinate loc3(0,255,0);
+        //loc2 (4,4,4)
```

Comment:
Would rather these be redefined than in a comment reiterating their value from earlier.